bigquery_etl/docs/derived_datasets/generate_derived_dataset_docs.py

```diff
@@ -68,6 +70,22 @@ def _get_referenced_tables_from_view(table_path):
     return referenced_tables
 
 
+def _get_schema(table_path):
+    """Get the (deployed) schema of the table."""
+    client = bigquery.Client()
+
+    project = table_path.parent.parent.name
+    dataset = table_path.parent.name
+    table = table_path.name
+    table_id = f"{project}.{dataset}.{table}"
+
+    try:
+        table = client.get_table(table_id)
+        return table.schema
+    except NotFound as e:
+        logging.warning(f"Unable to find schema. {e}")
```

Comment:
It would be nice to replace this with a call to the existing schema API (`Schema.for_table` for eg.) but dry-running doesn't return schema descriptions. Might be an idea to use the client directly instead of dry-running [here](https://github.com/mozilla/bigquery-etl/blob/main/bigquery_etl/schema/__init__.py#L55-L70) thoughts @scholtzan?

Reply:
Since we generate the docs as part of the CI we don't have access to a service account that can be used with the client directly.
We could think about extending the dry run function to return schema descriptions: https://github.com/mozilla-services/cloudops-infra/blob/master/projects/data-shared/tf/modules/cloudfunctions/src/bigquery_etl_dryrun/index.js

Reply:
Gotcha.. Yeah I was trying to avoid relying on referenced tables since any query other than `SELECT * FROM view` gets weird and we'll have to do name matching, etc. And I guess we really wouldn't want to add a service account to CI with limited permissions?

Reply:
Not with access to `shared-prod`. I guess there is a risk of accidentally exposing data. We do have a service account to the integration testing project, but that one doesn't have access to shared-prod, so not really helpful for this.

Reply:
I think we might be able to get creative with a custom IAM role and permissions that only allows listing and getting metadata if it's worth it.. Will put my thoughts down in an issue/proposal or something.

For now though, I think dry running is gonna require a couple changes in different places.. recording all the view partition keys plus the cloud function changes so I'll leave this as a draft for now